exporter/datadogexporter/metrics_exporter.go

```diff
@@ -20,20 +20,33 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
 	"gopkg.in/zorkian/go-datadog-api.v2"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/config"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metadata"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metrics"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/translator"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/utils"
 )
 
 type metricsExporter struct {
-	params  component.ExporterCreateSettings
-	cfg     *config.Config
-	ctx     context.Context
-	client  *datadog.Client
-	prevPts *ttlCache
+	params component.ExporterCreateSettings
+	cfg    *config.Config
+	ctx    context.Context
+	client *datadog.Client
+	tr     *translator.Translator
+}
+
+var _ translator.HostnameProvider = (*hostProvider)(nil)
```

Comment:
Maybe add a comment here to explain the purpose of this line

Reply:
It's an ['interface assertion'](https://stackoverflow.com/a/44976881) (there is no official name for it, really). It is a way for the compiler to typecheck that a given struct implements all methods of an interface